src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -0,0 +1,597 @@
+        version = "1.1.0")
+public class ValidateAIFCli implements Callable<Integer> {
+
+    // Program return codes from the AIF Validator.
+    public enum ReturnCode {
+        SUCCESS, VALIDATION_ERROR, USAGE_ERROR, FILE_ERROR
+    }
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Package Constants
+    // ----------------------------
+    // Error strings
+    static final String ERR_MISSING_ONT_FLAG = "Must use one of these flags: --ldc | --program | --ont";
+    static final String ERR_TOO_MANY_ONT_FLAGS = "Can only use one of these flags: --ldc | --program | --ont";
+    static final String ERR_MISSING_FILE_FLAG = "Must use one of these flags: -f | -d";
+    static final String ERR_TOO_MANY_FILE_FLAGS = "Can only use one of these flags: -f | -d";
+    static final String ERR_SMALLER_THAN_MIN = "%s must be at least %d";
+    // Logging strings
+    static final String START_MSG = "AIF Validator";
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Internal Constants
+    // ----------------------------
+    // Logger
+    private static final Logger logger = (Logger) (org.slf4j.LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME));
+    // ViolationThreshold
+    private static final String ABORT_PARAMETER_STRING = "Abort parameter";
+    private static final int DEFAULT_MAX_VIOLATIONS = -1;
+    private static final int MINIMUM_MAX_VIOLATIONS = 3;
+    // Profiling
+    private static final int LONG_QUERY_THRESH = 2000;
+    // Threading
+    private static final String THREAD_COUNT_STRING = "Thread count";
+    private static final int MINIMUM_THREAD_COUNT = 1;
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Command Line Arguments
+    // ----------------------------
+    private int argCounter = 1;
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "--ldc", description = "Validate against the LDC ontology")
+    private boolean useLDCOntology;
+
+    @Option(names = "--program", description = "Validate against the program ontology")
+    private boolean useProgramOntology;
+
+    @Option(names = "--ont", description = "Validate against the OWL-formatted ontolog(ies) at the specified filename(s)",
+            paramLabel = "FILE", arity = "1..*")
+    private List<File> customOntologies;
+
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "--nist", description = "Validate against the NIST restrictions")
+    private boolean useNISTRestriction;
+
+    @Option(names = "--nist-ta3", description = "Validate against the NIST hypothesis restrictions (implies --nist)")
+    private boolean useNISTTA3Rescriction;
+
+    @Option(names = "-o", description = "Save validation report model to a file.  KB.ttl would result in KB-report.txt.")
+    private boolean outputToFile;
+
+    @Option(names = "--abort", description = "Abort validation after [num] SHACL violations (num > 2), or 3 violations if [num] is omitted.", paramLabel = "num")
+    private int maxValidationErrors = DEFAULT_MAX_VIOLATIONS;
```

Comment:
The `[num]` parameter for `--abort` is supposed to be optional (and defaults to 3).
But `validateAIF --ldc -f foo.ttl --abort` results in `Missing required parameter for option '--abort' (num)`.
Similarly, `validateAIF --abort --ldc -f foo.ttl` should be valid.

Reply:
Fixed